exercises/concept/high-score-board/.meta/design.md

```diff
@@ -0,0 +1,94 @@
+# Design
+
+## Learning objectives
+
+- What are objects - for now describe them as maps/dictionaries that hold key-value pairs, more advanced things will be taught later
+- Objects are the core building blocks in JavaScript
+- How to create an object literal with `{}` (either empty or with some initial values)
+- What is allowed as keys
+- What is allowed as values (e.g., numbers, strings etc or other objects, arrays or even functions)
+- How to add a key-value pair to an object, how to change the value of an existing key
+- How to retrieve the value (show `obj["key"]` and `obj.key` notation)
+- How to retrieve nested values
+- How to remove an entry
+- How to check a key exists in the object with `hasOwnProperty`
+- How to iterate through the keys with `for ... in`
+
+## Out of Scope
+
+- Prototypes and classes
+- `this`
+- Object destructering
+- null and undefined (will be introduced in another concept exercise, including when they show up in the context of objects, it is a bit hard to tiptoe around this in this concept/exercise but the student can't learn everything at once)
+- `new Object()` / `Object.create`
+- Symbols as keys (should be covered in a separate concept)
+- JSON (should be covered in a separate concept)
+- "pass by reference" (will be introduced in the `functions` concept)
+- optional chaining (will be introduced in the `null-undefined` concept)
+
+## Concepts
+
+The Concepts this exercise unlocks are:
+
+- `objects`
+
+## Prerequisites
+
+- `for-loops` to better understand the for-in loop and because should already have covered some ground in the concept tree before starting with objects (e.g., arrays which are a prerequisite for `for-loops`)
+
+## Analyzer
+
+This exercise could benefit from the following rules in the [analyzer][analyzer]:
+
+1. `createScoreBoard`
+
+   - `essential`: Make sure no class, map etc. was created, there should be just an object.
+   - `actionable`: If the student created an empty object first and then added the value, give feedback to include the entry in the object literal directly.
+   - `actionable`: Check that the object was returned directly, no intermediate assignment to a variable necessary.
+
+2. `addPlayer`
+
+   - ???
```

Comment:
```suggestion
   - TODO/FIXME
```

Again, not sure what the conventions are in the JS repo, but `???` isn't easily findable via grepping, I'm guessing.

Reply:
true, but I had in idea what to write there now so this became obsolete